include/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth/NRLMSISE00.hpp

```diff
@@ -0,0 +1,155 @@
+#include <OpenSpaceToolkit/Physics/Time/Instant.hpp>
+#include <OpenSpaceToolkit/Physics/Units/Length.hpp>
+
+namespace ostk
+{
+namespace physics
+{
+namespace environment
+{
+namespace atmospheric
+{
+namespace earth
+{
+
+using ostk::core::types::Integer;
+using ostk::core::types::Real;
+using ostk::core::types::String;
+using ostk::core::ctnr::Tuple;
+using ostk::core::ctnr::Array;
+
+using ostk::physics::time::Instant;
+using ostk::physics::units::Length;
+using ostk::physics::coord::spherical::LLA;
+using ostk::physics::environment::atmospheric::Model;
+
+/// @brief                      NRLMSISE00 atmospheric model
+
+class NRLMSISE00 : public Model
+{
+   public:
+    /// @brief              Constructor
+
+    NRLMSISE00();
+
+    /// @brief              Clone the exponential atmospheric model
+    ///
+    /// @return             Pointer to exponential atmospheric model
+
+    virtual NRLMSISE00* clone() const override;
+
+    /// @brief              Check if the exponential atmospheric model is defined
+    ///
+    /// @return             True if the exponential atmospheric model is defined
+
+    bool isDefined() const;
+
+    /// @brief              Get the atmospheric density value at a given position and instant.
+    ///                     Optionally use provided sun position to calculate local solar time.
+    ///
+    /// @param              [in] aLLA A position, expressed as latitude, longitude, altitude [deg, deg, m]
+    /// @param              [in] anInstant An instant
+    /// @param              [in] aSunPosition Position of the sun
+    /// @return             Atmospheric density value [kg.m^-3]
+
+    Real getDensityAt(const LLA& aLLA, const Instant& anInstant, const Position& aSunPosition = Position::Undefined())
+        const;
+
+    /// @brief              Get the atmospheric density value at a given position and instant
+    ///                     Use provided sun position to calculate local solar time.
+    ///
+    /// @param              [in] aPosition A Position
+    /// @param              [in] anInstant An Instant
+    /// @param              [in] aSunPosition Position of the sun
+    /// @return             Atmospheric density value [kg.m^-3]
+
+    Real getDensityAt(const Position& aPosition, const Instant& anInstant, const Position& aSunPosition) const;
+
+    /// @brief              Get the atmospheric density value at a given position and instant.
+    ///                     Use lst = secondsInDay/3600.0 + aLLA.getLongitude().inDegrees()/15.0 to calculate local
+    ///                     solar time.
+    ///
+    /// @param              [in] aLLA A position, expressed as latitude, longitude, altitude [deg, deg, m]
+    /// @param              [in] anInstant An instant
+    /// @return             Atmospheric density value [kg.m^-3]
+
+    virtual Real getDensityAt(const Position& aPosition, const Instant& anInstant) const override;
+
+   protected:
```

Comment:
These are protected to avoid exposing them in the interface. The c array/struct thing is gross.
protected instead of private so that they can be accessed in the tests.